### PR TITLE
feat(api): tracing and performance improvements

### PR DESCRIPTION
**Pull request by fsamin**
1. Description
1. Related issues
1. About tests
1. Mentions
@ovh/cds

### CODE REVIEW

File: engine/api/tracing/helper.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
+package tracing
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/go-gorp/gorp"
```
Comment (sguiheux) [inline engine/api/tracing/helper.go]: import order

File: engine/api/tracing/tracing.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	"github.com/go-gorp/gorp"
+	"go.opencensus.io/exporter/jaeger"
+	"go.opencensus.io/trace"
+
+	"github.com/ovh/cds/engine/api/cache"
+	"github.com/ovh/cds/engine/api/feature"
+	"github.com/ovh/cds/sdk/log"
+)
+
+/* Start jarger with:
```
Comment (sguiheux) [inline engine/api/tracing/tracing.go]: remove

File: engine/api/project/dao.go [hunk 3]
```diff
@@ -60,7 +62,11 @@ func LoadAllByRepo(db gorp.SqlExecutor, store cache.Store, u *sdk.User, repo str
 }
 
 // LoadAll returns all projects
-func LoadAll(db gorp.SqlExecutor, store cache.Store, u *sdk.User, opts ...LoadOptionFunc) ([]sdk.Project, error) {
+func LoadAll(ctx context.Context, db gorp.SqlExecutor, store cache.Store, u *sdk.User, opts ...LoadOptionFunc) ([]sdk.Project, error) {
+	var end func()
+	ctx, end = tracing.Span(ctx, "project.LoadAll")
```
Comment (golangci) [inline engine/api/project/dao.go]: ineffectual assignment to `ctx`

File: engine/api/project/dao.go [hunk 4]
```diff
@@ -390,7 +396,11 @@ func loadprojects(db gorp.SqlExecutor, store cache.Store, u *sdk.User, opts []Lo
 	return projs, nil
 }
 
-func load(db gorp.SqlExecutor, store cache.Store, u *sdk.User, opts []LoadOptionFunc, query string, args ...interface{}) (*sdk.Project, error) {
+func load(ctx context.Context, db gorp.SqlExecutor, store cache.Store, u *sdk.User, opts []LoadOptionFunc, query string, args ...interface{}) (*sdk.Project, error) {
+	var end func()
+	ctx, end = tracing.Span(ctx, "project.load")
```
Comment (golangci) [inline engine/api/project/dao.go]: ineffectual assignment to `ctx`

File: engine/api/workflow/dao_node_run.go [hunk 5]
```diff
@@ -86,7 +88,11 @@ func LoadNodeRun(db gorp.SqlExecutor, projectkey, workflowname string, number, i
 }
 
 //LoadAndLockNodeRunByID load and lock a specific node run on a workflow
-func LoadAndLockNodeRunByID(db gorp.SqlExecutor, id int64, wait bool) (*sdk.WorkflowNodeRun, error) {
+func LoadAndLockNodeRunByID(ctx context.Context, db gorp.SqlExecutor, id int64, wait bool) (*sdk.WorkflowNodeRun, error) {
+	var end func()
+	ctx, end = tracing.Span(ctx, "workflow.LoadAndLockNodeRunByID")
```
Comment (golangci) [inline engine/api/workflow/dao_node_run.go]: ineffectual assignment to `ctx`

File: engine/api/workflow_queue.go [hunk 6]
```diff
@@ -96,39 +102,35 @@ func (api *API) postTakeWorkflowJobHandler() Handler {
...
 		}
 
-		chanEvent := make(chan interface{}, 1)
-		chanError := make(chan error, 1)
-
 		pbji := &worker.WorkflowNodeJobRunInfo{}
-		go takeJob(ctx, chanEvent, chanError, api.mustDB(), api.Cache, p, getWorker(ctx), id, takeForm, workerModel, pbji)
 
-		workflowRuns, workflowNodeRuns, workflowNodeJobRuns, err := workflow.GetWorkflowRunEventData(chanError, chanEvent, p.Key)
+		report, err := takeJob(ctx, api.mustDB(), api.Cache, p, getWorker(ctx), id, takeForm, workerModel, pbji)
```
Comment (golangci) [inline engine/api/workflow_queue.go]: ineffectual assignment to `err`

File: engine/api/workflow_run.go [hunk 7]
```diff
@@ -672,22 +679,8 @@ func (api *API) postWorkflowRunHandler() Handler {
...
-					runtime.Stack(buf, true)
-					log.Error("[PANIC] workflow.startWorkflowRun> %s", string(buf))
-				}
-			}()
-
-			startWorkflowRun(chanEvent, chanError, api.mustDB(), api.Cache, p, wf, lastRun, opts, u, asCodeInfosMsg)
-		}()
-
-		workflowRuns, workflowNodeRuns, workflowNodeJobRuns, err := workflow.GetWorkflowRunEventData(chanError, chanEvent, p.Key)
+		report, err := startWorkflowRun(ctx, api.mustDB(), api.Cache, p, wf, lastRun, opts, u, asCodeInfosMsg)
```
Comment (golangci) [inline engine/api/workflow_run.go]: ineffectual assignment to `err`

File: sdk/cdsclient/http.go [hunk 8]
```diff
@@ -39,6 +41,8 @@ const (
 	ResponseProcessTimeHeader = "X-Api-Process-Time"
 )
 
+var defaultFormat propagation.HTTPFormat = &b3.HTTPFormat{}
```
Comment (golangci) [inline sdk/cdsclient/http.go]: `defaultFormat` is unused

File: engine/api/tracing/tracker.go [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+)
+
+func spanContextToReponse(ctx trace.SpanContext, r *http.Request, w http.ResponseWriter) {
+	w.Header().Add(tracingutils.TraceIDHeader, hex.EncodeToString(ctx.TraceID[:]))
+	w.Header().Add(tracingutils.SpanIDHeader, hex.EncodeToString(ctx.SpanID[:]))
+	w.Header().Add(tracingutils.SampledHeader, r.Header.Get(tracingutils.SampledHeader))
+}
+
+// spanContextToRequest writes span context to http requests
+func spanContextToRequest(ctx trace.SpanContext, r *http.Request) {
```
Comment (golangci) [inline engine/api/tracing/tracker.go]: `spanContextToRequest` is unused

File: engine/api/workflow/execute_node_run.go [hunk 10]
```diff
@@ -196,37 +203,40 @@ func execute(dbCopy *gorp.DbMap, db gorp.SqlExecutor, store cache.Store, proj *s
...
+		report.Add(*n)
 		if n.Status != sdk.StatusStopped.String() {
-			if _, err := processWorkflowRun(dbCopy, db, store, proj, updatedWorkflowRun, nil, nil, nil, chanEvent); err != nil {
-				return sdk.WrapError(err, "workflow.execute> Unable to reprocess workflow !")
+
+			r1, _, err := processWorkflowRun(ctx, dbCopy, db, store, proj, updatedWorkflowRun, nil, nil, nil)
+			if err != nil {
+				return report, sdk.WrapError(err, "workflow.execute> Unable to reprocess workflow !")
 			}
+			report.Merge(r1, nil)
```
Comment (golangci) [inline engine/api/workflow/execute_node_run.go]: Error return value of `report.Merge` is not checked

File: engine/api/workflow/execute_node_run.go [hunk 11]
```diff
@@ -569,20 +598,28 @@ func NodeBuildParametersFromWorkflow(db gorp.SqlExecutor, store cache.Store, pro
...
+		return report, sdk.WrapError(errIDS, "StopWorkflowNodeRun> Cannot load node jobs run ids ")
 	}
 
 	chanNjrID := make(chan int64, stopWorkflowNodeRunNBWorker)
 	chanNodeJobRunDone := make(chan bool, stopWorkflowNodeRunNBWorker)
 	chanErr := make(chan error, stopWorkflowNodeRunNBWorker)
 	for i := 0; i < stopWorkflowNodeRunNBWorker && i < len(ids); i++ {
-		go stopWorkflowNodeJobRun(db, store, proj, &nodeRun, stopInfos, chanNjrID, chanEvent, chanErr, chanNodeJobRunDone, &wg)
+		go func() {
+			report.Merge(stopWorkflowNodeJobRun(ctx, db, store, proj, &nodeRun, stopInfos, chanNjrID, chanErr, chanNodeJobRunDone, &wg), nil)
```
Comment (golangci) [inline engine/api/workflow/execute_node_run.go]: Error return value of `report.Merge` is not checked

File: engine/api/workflow/process.go [hunk 12]
```diff
@@ -57,16 +64,17 @@ func processWorkflowRun(dbCopy *gorp.DbMap, db gorp.SqlExecutor, store cache.Sto
...
+				return report, false, sdk.ErrWorkflowNodeParentNotRun
 			}
 		}
-		conditionOK, errP := processWorkflowNodeRun(dbCopy, db, store, proj, w, start, int(nextSubNumber), sourceNodesRunID, hookEvent, manual, chanEvent)
+		r1, conditionOK, errP := processWorkflowNodeRun(ctx, dbCopy, db, store, proj, w, start, int(nextSubNumber), sourceNodesRunID, hookEvent, manual)
 		if errP != nil {
-			return false, sdk.WrapError(errP, "processWorkflowRun> Unable to process workflow node run")
+			return report, false, sdk.WrapError(errP, "processWorkflowRun> Unable to process workflow node run")
 		}
+		report.Merge(r1, nil)
```
Comment (golangci) [inline engine/api/workflow/process.go]: Error return value of `report.Merge` is not checked

File: engine/api/workflow_run.go [hunk 13]
```diff
@@ -715,181 +708,149 @@ func (api *API) postWorkflowRunHandler() Handler {
...
+	tx, errb := db.Begin()
+	if errb != nil {
+		return nil, errb
+	}
 
-		//If payload is not set, keep the default payload
-		if opts.Manual.Payload == interface{}(nil) {
-			opts.Manual.Payload = fromNode.Context.DefaultPayload
-		}
+	defer tx.Rollback()
```
Comment (golangci) [inline engine/api/workflow_run.go]: Error return value of `tx.Rollback` is not checked

File: engine/api/workflow/execute_node_job_run.go [hunk 14]
```diff
@@ -15,36 +17,97 @@ import (
...
-// UpdateNodeJobRunStatus Update status of an workflow_node_run_job
-func UpdateNodeJobRunStatus(dbCopy *gorp.DbMap, db gorp.SqlExecutor, store cache.Store, proj *sdk.Project, job *sdk.WorkflowNodeJobRun, status sdk.Status, chanEvent chan<- interface{}) error {
-	log.Debug("UpdateNodeJobRunStatus> job.ID=%d status=%s", job.ID, status.String())
+// ProcessorReport represents the state of the workflow processor
+type ProcessorReport struct {
+	mutex     sync.Mutex
+	jobs      []sdk.WorkflowNodeJobRun
+	nodes     []sdk.WorkflowNodeRun
+	workflows []sdk.WorkflowRun
+	errors    []error
```
Comment (golangci) [inline engine/api/workflow/execute_node_job_run.go]: `errors` is unused

File: engine/api/tracing/types.go [hunk 15]
```diff
@@ -0,0 +1,41 @@
+package tracing
+
+import (
```
Comment (sguiheux) [inline engine/api/tracing/types.go]: order

File: sdk/cdsclient/http.go [hunk 16]
```diff
@@ -13,6 +13,8 @@ import (
 	"strings"
 
 	"github.com/ovh/cds/sdk"
+	"go.opencensus.io/plugin/ochttp/propagation/b3"
```
Comment (sguiheux) [inline sdk/cdsclient/http.go]: order

File: engine/api/workflow/dao_node_run_job.go [hunk 17]
```diff
@@ -148,7 +150,11 @@ func LoadAndLockNodeJobRunWait(db gorp.SqlExecutor, store cache.Store, id int64)
 }
 
 //LoadAndLockNodeJobRunNoWait load for update a NodeJobRun given its ID
-func LoadAndLockNodeJobRunNoWait(db gorp.SqlExecutor, store cache.Store, id int64) (*sdk.WorkflowNodeJobRun, error) {
+func LoadAndLockNodeJobRunNoWait(ctx context.Context, db gorp.SqlExecutor, store cache.Store, id int64) (*sdk.WorkflowNodeJobRun, error) {
+	var end func()
+	ctx, end = tracing.Span(ctx, "workflow.LoadAndLockNodeJobRunNoWait")
```
Comment (golangci) [inline engine/api/workflow/dao_node_run_job.go]: ineffectual assignment to `ctx`

File: engine/api/workflow/dao_node_run_job.go [hunk 18]
```diff
@@ -176,7 +182,11 @@ func DeleteNodeJobRuns(db gorp.SqlExecutor, nodeID int64) error {
 }
 
 //UpdateNodeJobRun updates a workflow_node_run_job
-func UpdateNodeJobRun(db gorp.SqlExecutor, store cache.Store, j *sdk.WorkflowNodeJobRun) error {
+func UpdateNodeJobRun(ctx context.Context, db gorp.SqlExecutor, store cache.Store, j *sdk.WorkflowNodeJobRun) error {
+	var end func()
+	ctx, end = tracing.Span(ctx, "workflow.UpdateNodeJobRun")
```
Comment (golangci) [inline engine/api/workflow/dao_node_run_job.go]: ineffectual assignment to `ctx`

File: engine/api/workflow/dao_run.go [hunk 19]
```diff
@@ -46,7 +48,11 @@ func insertWorkflowRun(db gorp.SqlExecutor, wr *sdk.WorkflowRun) error {
 }
 
 // UpdateWorkflowRun updates in table "workflow_run""
-func UpdateWorkflowRun(db gorp.SqlExecutor, wr *sdk.WorkflowRun) error {
+func UpdateWorkflowRun(ctx context.Context, db gorp.SqlExecutor, wr *sdk.WorkflowRun) error {
+	var end func()
+	ctx, end = tracing.Span(ctx, "workflow.UpdateWorkflowRun")
```
Comment (golangci) [inline engine/api/workflow/dao_run.go]: ineffectual assignment to `ctx`

File: engine/api/workflow/process.go [hunk 20]
```diff
@@ -81,11 +89,12 @@ func processWorkflowRun(dbCopy *gorp.DbMap, db gorp.SqlExecutor, store cache.Sto
...
 		})
 
-		conditionOK, errP := processWorkflowNodeRun(dbCopy, db, store, proj, w, w.Workflow.Root, 0, nil, hookEvent, manual, chanEvent)
+		r1, conditionOK, errP := processWorkflowNodeRun(ctx, dbCopy, db, store, proj, w, w.Workflow.Root, 0, nil, hookEvent, manual)
 		if errP != nil {
-			return false, sdk.WrapError(errP, "processWorkflowRun> Unable to process workflow node run")
+			return report, false, sdk.WrapError(errP, "processWorkflowRun> Unable to process workflow node run")
 		}
-		return conditionOK, nil
+		report.Merge(r1, nil)
```
Comment (golangci) [inline engine/api/workflow/process.go]: Error return value of `report.Merge` is not checked

File: engine/api/workflow/process.go [hunk 21]
```diff
@@ -126,14 +135,16 @@ func processWorkflowRun(dbCopy *gorp.DbMap, db gorp.SqlExecutor, store cache.Sto
...
-						conditionOk, errPwnr := processWorkflowNodeRun(dbCopy, db, store, proj, w, &t.WorkflowDestNode, int(nodeRun.SubNumber), []int64{nodeRun.ID}, nil, nil, chanEvent)
+						r1, conditionOk, errPwnr := processWorkflowNodeRun(ctx, dbCopy, db, store, proj, w, &t.WorkflowDestNode, int(nodeRun.SubNumber), []int64{nodeRun.ID}, nil, nil)
 						if errPwnr != nil {
 							log.Error("processWorkflowRun> Unable to process node ID=%d: %s", t.WorkflowDestNode.ID, errPwnr)
 							AddWorkflowRunInfo(w, true, sdk.SpawnMsg{
 								ID:   sdk.MsgWorkflowError.ID,
 								Args: []interface{}{errPwnr.Error()},
 							})
 						}
+						report.Merge(r1, nil)
```
Comment (golangci) [inline engine/api/workflow/process.go]: Error return value of `report.Merge` is not checked

File: engine/api/workflow/process.go [hunk 22]
```diff
@@ -244,14 +255,15 @@ func processWorkflowRun(dbCopy *gorp.DbMap, db gorp.SqlExecutor, store cache.Sto
...
-					conditionOK, err := processWorkflowNodeRun(dbCopy, db, store, proj, w, &t.WorkflowDestNode, int(maxsn), nodeRunIDs, nil, nil, chanEvent)
+					r1, conditionOK, err := processWorkflowNodeRun(ctx, dbCopy, db, store, proj, w, &t.WorkflowDestNode, int(maxsn), nodeRunIDs, nil, nil)
 					if err != nil {
 						AddWorkflowRunInfo(w, true, sdk.SpawnMsg{
 							ID:   sdk.MsgWorkflowError.ID,
 							Args: []interface{}{err.Error()},
 						})
 						log.Error("processWorkflowRun> Unable to process node ID=%d: %v", t.WorkflowDestNode.ID, err)
 					}
+					report.Merge(r1, nil)
```
Comment (golangci) [inline engine/api/workflow/process.go]: Error return value of `report.Merge` is not checked

File: engine/api/workflow/execute_node_run.go [hunk 23]
```diff
@@ -625,57 +663,68 @@ func StopWorkflowNodeRun(db *gorp.DbMap, store cache.Store, proj *sdk.Project, n
...
 		wg.Done()
 	}
+	return report
 }
 
 // SyncNodeRunRunJob sync step status and spawnInfos in a specific run job
-func SyncNodeRunRunJob(db gorp.SqlExecutor, nodeRun *sdk.WorkflowNodeRun, nodeJobRun sdk.WorkflowNodeJobRun) (bool, error) {
+func SyncNodeRunRunJob(ctx context.Context, db gorp.SqlExecutor, nodeRun *sdk.WorkflowNodeRun, nodeJobRun sdk.WorkflowNodeJobRun) (bool, error) {
+	var end func()
+	ctx, end = tracing.Span(ctx, "workflow.SyncNodeRunRunJob")
```
Comment (golangci) [inline engine/api/workflow/execute_node_run.go]: ineffectual assignment to `ctx`

File: engine/api/workflow/process.go [hunk 24]
```diff
@@ -651,22 +666,23 @@ func processWorkflowNodeRun(dbCopy *gorp.DbMap, db gorp.SqlExecutor, store cache
...
 	//Execute the node run !
-	if err := execute(dbCopy, db, store, p, run, chanEvent); err != nil {
-		return true, sdk.WrapError(err, "processWorkflowNodeRun> unable to execute workflow run")
+	r1, err := execute(ctx, dbCopy, db, store, p, run)
+	if err != nil {
+		return report, true, sdk.WrapError(err, "processWorkflowNodeRun> unable to execute workflow run")
 	}
-
-	return true, nil
+	report.Merge(r1, nil)
```
Comment (golangci) [inline engine/api/workflow/process.go]: Error return value of `report.Merge` is not checked

File: engine/api/workflow/run_workflow.go [hunk 25]
```diff
@@ -66,25 +74,26 @@ func RunFromHook(dbCopy *gorp.DbMap, db gorp.SqlExecutor, store cache.Store, p *
...
 		}
 
 		//Process it
-		hasRun, errWR := processWorkflowRun(dbCopy, db, store, p, wr, e, nil, nil, chanEvent)
+		r1, hasRun, errWR := processWorkflowRun(ctx, dbCopy, db, store, p, wr, e, nil, nil)
 		if errWR != nil {
-			return nil, sdk.WrapError(errWR, "RunFromHook> Unable to process workflow run")
+			return nil, report, sdk.WrapError(errWR, "RunFromHook> Unable to process workflow run")
 		}
+		report.Merge(r1, nil)
```
Comment (golangci) [inline engine/api/workflow/run_workflow.go]: Error return value of `report.Merge` is not checked

File: engine/api/workflow/run_workflow.go [hunk 26]
```diff
@@ -93,56 +102,65 @@ func RunFromHook(dbCopy *gorp.DbMap, db gorp.SqlExecutor, store cache.Store, p *
...
 		}
 
 		//Process the workflow run from the node ID
-		if _, err := processWorkflowRun(dbCopy, db, store, p, lastWorkflowRun, e, nil, &oldH.WorkflowNodeID, chanEvent); err != nil {
-			return nil, sdk.WrapError(err, "RunFromHook> Unable to process workflow run")
+		r1, _, err := processWorkflowRun(ctx, dbCopy, db, store, p, lastWorkflowRun, e, nil, &oldH.WorkflowNodeID)
+		if err != nil {
+			return nil, report, sdk.WrapError(err, "RunFromHook> Unable to process workflow run")
 		}
+		report.Merge(r1, nil)
```
Comment (golangci) [inline engine/api/workflow/run_workflow.go]: Error return value of `report.Merge` is not checked

File: engine/api/workflow/run_workflow.go [hunk 27]
```diff
@@ -93,56 +102,65 @@ func RunFromHook(dbCopy *gorp.DbMap, db gorp.SqlExecutor, store cache.Store, p *
...
 	}
 	lastWorkflowRun.Tag(tagTriggeredBy, e.User.Username)
 
-	condOk, err := processWorkflowRun(dbCopy, db, store, p, lastWorkflowRun, nil, e, &nodeID, chanEvent)
+	r1, condOk, err := processWorkflowRun(ctx, dbCopy, db, store, p, lastWorkflowRun, nil, e, &nodeID)
 	if err != nil {
-		return nil, sdk.WrapError(err, "ManualRunFromNode> Unable to process workflow run")
+		return nil, report, sdk.WrapError(err, "ManualRunFromNode> Unable to process workflow run")
 	}
+	report.Merge(r1, nil)
```
Comment (golangci) [inline engine/api/workflow/run_workflow.go]: Error return value of `report.Merge` is not checked

File: engine/api/workflow/run_workflow.go [hunk 28]
```diff
@@ -159,16 +177,17 @@ func ManualRun(dbCopy *gorp.DbMap, db gorp.SqlExecutor, store cache.Store, p *sd
...
+		return nil, report, sdk.WrapError(err, "ManualRun> Unable to manually run workflow %s/%s", w.ProjectKey, w.Name)
 	}
 
-	hasRun, errWR := processWorkflowRun(dbCopy, db, store, p, wr, nil, e, nil, chanEvent)
+	r1, hasRun, errWR := processWorkflowRun(ctx, dbCopy, db, store, p, wr, nil, e, nil)
 	if errWR != nil {
-		return wr, sdk.WrapError(errWR, "ManualRun")
+		return wr, report, sdk.WrapError(errWR, "ManualRun")
 	}
+	report.Merge(r1, nil)
```
Comment (golangci) [inline engine/api/workflow/run_workflow.go]: Error return value of `report.Merge` is not checked

File: engine/api/tracing/helper.go (hunk 1 above)
Comment (fsamin) [reply engine/api/tracing/helper.go]: done

File: engine/api/tracing/types.go (hunk 15 above)
Comment (fsamin) [reply engine/api/tracing/types.go]: done

File: engine/api/workflow_run.go [hunk 29]
```diff
@@ -367,26 +364,28 @@ func stopWorkflowRun(chEvent chan<- interface{}, chError chan<- error, db *gorp.
...
 			}
 
-			if errS := workflow.StopWorkflowNodeRun(db, store, p, wnr, stopInfos, chEvent); errS != nil {
-				chError <- sdk.WrapError(errS, "stopWorkflowRunHandler> Unable to stop workflow node run %d", wnr.ID)
+			r1, errS := workflow.StopWorkflowNodeRun(ctx, db, store, p, wnr, stopInfos)
+			if errS != nil {
+				return nil, sdk.WrapError(errS, "stopWorkflowRunHandler> Unable to stop workflow node run %d", wnr.ID)
 				tx.Rollback()
 			}
+			report.Merge(r1, nil)
```
Comment (golangci) [inline engine/api/workflow_run.go]: Error return value of `report.Merge` is not checked

File: engine/api/workflow_run.go [hunk 30]
```diff
@@ -551,25 +543,28 @@ func stopWorkflowNodeRun(chEvent chan<- interface{}, chError chan<- error, db *g
...
 
-	if errR := workflow.ResyncWorkflowRunStatus(tx, wr, chEvent); errR != nil {
-		chError <- sdk.WrapError(errR, "stopWorkflowNodeRunHandler> Unable to resync workflow run status")
-		return
+	r1, errR := workflow.ResyncWorkflowRunStatus(tx, wr)
+	if errR != nil {
+		return nil, sdk.WrapError(errR, "stopWorkflowNodeRunHandler> Unable to resync workflow run status")
 	}
 
+	report.Merge(r1, nil)
```
Comment (golangci) [inline engine/api/workflow_run.go]: Error return value of `report.Merge` is not checked

File: engine/api/workflow_run.go [hunk 31]
```diff
@@ -715,181 +702,151 @@ func (api *API) postWorkflowRunHandler() Handler {
...
-		if err := tx.Commit(); err != nil {
-			chError <- sdk.WrapError(err, "postWorkflowRunHandler> Unable to commit transaction")
+	wg.Wait()
+
+	if lastRun == nil {
+		_, r1, errmr := workflow.ManualRun(ctx, db, tx, store, p, wf, opts.Manual, asCodeInfos)
+		if errmr != nil {
+			return nil, sdk.WrapError(errmr, "postWorkflowRunHandler> Unable to run workflow")
 		}
+		report.Merge(r1, nil)
```
Comment (golangci) [inline engine/api/workflow_run.go]: Error return value of `report.Merge` is not checked

**Pull request merged**